sklearn/covariance/robust_covariance.py

```diff
@@ -144,6 +144,16 @@ def _c_step(X, n_support, random_state, remaining_iterations=30,
         location = X_support.mean(axis=0)
         covariance = cov_computation_method(X_support)
         det = fast_logdet(covariance)
+
+        if det > previous_det:
+            # determinant has increased (should not happen)
+            warnings.warn("Determinant has increased; this should not happen: "
+                          "log(det) > log(previous_det) (%.15f > %.15f). "
+                          "You may want to try with a higher value of "
+                          "support_fraction (current value: %.3f)."
+                          % (det, previous_det, n_support / n_samples),
```

Comment:
fraction may be lost on python 2?

Reply:
I added `from __future__ import division`